test/WebJobs.Script.Tests/Security/Authentication/ArmAuthenticationHandlerTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Script.Tests.Helpers;
+using Microsoft.Azure.WebJobs.Script.WebHost.Security;
+using Microsoft.Azure.WebJobs.Script.WebHost.Security.Authentication;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Security.Authentication
+{
+    public class ArmAuthenticationHandlerTests
+    {
+        private static DefaultHttpContext GetContext()
+        {
+            var services = new ServiceCollection().AddLogging();
+            services.AddAuthentication(o =>
+            {
+                o.DefaultScheme = ArmAuthenticationDefaults.AuthenticationScheme;
+            })
+            .AddArmToken();
+
+            var sp = services.BuildServiceProvider();
+            var context = new DefaultHttpContext
+            {
+                RequestServices = sp
+            };
+            return context;
+        }
+
+        [Fact]
+        public async Task AuthenticateAsync_WithoutToken_DoesNotAuthenticate()
+        {
+            // Arrange
+            DefaultHttpContext context = GetContext();
+
+            // Act
+            AuthenticateResult result = await context.AuthenticateAsync();
+
+            // Assert
+            Assert.True(result.None);
+            Assert.Null(result.Failure);
+            Assert.Null(result.Principal);
+        }
+
+        [Fact]
+        public async Task AuthenticateAsync_WithToken_PerformsAuthentication()
+        {
+            using (new TestScopedEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey, SimpleWebTokenTests.GenerateKeyHexString()))
```

Comment:
this is neat, I didn't know you had that. I should change the other tests I added to use this instead